### DISCUSSION TITLE
Support filtering on unexisting fields

### DISCUSSION THREAD

**Discussion by MarinPostma** [meilisearch/product#89] [Feedback & Feature Proposal]
I realised that documents that don't have a value for a specific field are discarded. The question arises as how to handle them? Is an empty value the same thing as an unexisting field? I think not, but again this is debatable.

On a semantic level, the empty string is the only way we have to tell that we expect an empty value. 

```SQL
# this can't be a valid syntax
color = OR color = green
```
And the fact that we allow unquoted word a a value prevents us from adding a `null` keyword (which I'm against anyway).

Now we could have the following

```SQL
# matches all the documents that do not have a field `color`
NOT color
# matches all the documents that have a field `color`
color
```

but that implies a consequent change in the syntax.

**Comment by vmitchell85** [comment]
+1 I would also like this feature.... I ended up moving from MeiliSearch because I couldn't do this.

**Comment by gmourier** [comment]
Hey 👋

This may interest you @vmitchell85 https://github.com/meilisearch/product/discussions/407#discussioncomment-3084053! A prototype adding 2 new operators is available. It adds `IN` and `EXISTS`.

**Comment by gmourier** [comment]
Hey everyone 👋

I'm locking this discussion in favor of https://github.com/meilisearch/product/discussions/603

Thanks!
